parser-typechecker/unison/Compat.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE CPP #-}
```

Comment:
Splitting this off wasn't strictly necessary, but CPP messes with tooling sometimes and in my experience I've generally found that it's best to wall off any uses of CPP into a dark corner where it can't spread 😬 .

It also makes `Main.hs` less scary for any potential new contributors 😄